src/opensles/AudioStreamOpenSLES.cpp

```diff
@@ -163,6 +163,49 @@ PerformanceMode AudioStreamOpenSLES::convertPerformanceMode(SLuint32 openslMode)
     return oboeMode;
 }
 
+void AudioStreamOpenSLES::logUnsupportedAttributes() {
+    // Log unsupported attributes
+    // only report if changed from the default
+
+    // Device ID
+    if (mDeviceId != kUnspecified) {
+        LOGW("Device ID [AudioStreamBuilder::setDeviceId()] "
+             "is not supported on OpenSLES streams.");
+    }
+    // Sharing Mode
+    if (mSharingMode != SharingMode::Shared) {
+        LOGW("SharingMode [AudioStreamBuilder::setSharingMode()] "
+             "is not supported on OpenSLES streams.");
+    }
+    // Performance Mode
+    if (mPerformanceMode != PerformanceMode::None) {
+        LOGW("PerformanceMode [AudioStreamBuilder::setPerformanceMode()] "
+             "is not supported on OpenSLES streams.");
+    }
+    // Usage
+    if (mUsage != Usage::Media) {
```

Comment:
Usage is supported by OpenSL ES.
https://github.com/google/oboe/blob/6359dca09f7c91ebd8b92ffcc5738b041f685d12/src/opensles/AudioOutputStreamOpenSLES.cpp#L192